aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -62,7 +62,7 @@ static inline Tensor unary_float_op_impl(const Tensor& self, OutImpl& out_impl)
   Tensor result = isIntegralType(self.scalar_type(), /*include_bool=*/ true) ? \
                   at::empty({0}, self.options().dtype(typeMetaToScalarType(c10::get_default_dtype()))) : \
                   at::empty({0}, self.options());
-  return out_impl(result, self.to(result.scalar_type()));
+  return out_impl(result, self);
```

Comment:
The current implementation (using `PROMOTE` strategy for promotion in TensorIterator API for unary floating ufuncs) avoids `self.to(result.scalar_type())` call.

Reply:
This approach seems a bit confusing to me. I can see why it works, but it works for an odd reason.

By relying on regular promotion logic inside tensor iterator, we're depending on the fact that `result_type(input_tensor, desired_result) == desired_result` for any possible dtype. This seems to work -- `torch.result_type(int_tensor, float32_tensor)` gives us float32 not float64... but this is relying on the table in ScalarType.h to get the result, not actually relying directly on the call to `torch.get_default_dtype`. 

I think I'd prefer to be more explicit in how we handle this, to simplify any changes that might be needed for additional dtypes later.